Sources/Sentry/SentryReachability.m

```diff
@@ -0,0 +1,155 @@
+    // kSCNetworkReachabilityFlagsIsWWAN does not exist on macOS
+    const SCNetworkReachabilityFlags importantFlags
+        = kSCNetworkReachabilityFlagsIsWWAN | kSCNetworkReachabilityFlagsReachable;
+#    else
+    const SCNetworkReachabilityFlags importantFlags = kSCNetworkReachabilityFlagsReachable;
+#    endif
+    __block BOOL shouldReport = YES;
+    // Check if the reported state is different from the last known state (if any)
+    SCNetworkReachabilityFlags newFlags = flags & importantFlags;
+    SCNetworkReachabilityFlags oldFlags = sentry_current_reachability_state & importantFlags;
+    if (newFlags != oldFlags) {
+        // When first subscribing to be notified of changes, the callback is
+        // invoked immmediately even if nothing has changed. So this block
+        // ignores the very first check, reporting all others.
+        if (sentry_current_reachability_state == kSCNetworkReachabilityFlagsUninitialized) {
+            shouldReport = NO;
+        }
+        // Cache the reachability state to report the previous value representation
+        sentry_current_reachability_state = flags;
+    } else {
+        shouldReport = NO;
+    }
+    return shouldReport;
+}
+
+/**
+ * Textual representation of a connection type
+ */
+NSString *
+SentryConnectivityFlagRepresentation(SCNetworkReachabilityFlags flags)
+{
+    BOOL connected = (flags & kSCNetworkReachabilityFlagsReachable) != 0;
+#    if SENTRY_HAS_UIDEVICE
+    return connected ? ((flags & kSCNetworkReachabilityFlagsIsWWAN) ? SentryConnectivityCellular
+                                                                    : SentryConnectivityWiFi)
+                     : SentryConnectivityNone;
+#    else
+    return connected ? SentryConnectivityWiFi : SentryConnectivityNone;
+#    endif
+}
+
+/**
+ * Callback invoked by SCNetworkReachability, which calls an Objective-C block
+ * that handles the connection change.
+ */
+void
+SentryConnectivityCallback(
+    __unused SCNetworkReachabilityRef target, SCNetworkReachabilityFlags flags, __unused void *info)
+{
+    if (sentry_reachability_change_block && SentryConnectivityShouldReportChange(flags)) {
+        BOOL connected = (flags & kSCNetworkReachabilityFlagsReachable) != 0;
+        sentry_reachability_change_block(connected, SentryConnectivityFlagRepresentation(flags));
+    }
+}
+
+@implementation SentryReachability
+
++ (void)monitorURL:(NSURL *)URL usingCallback:(SentryConnectivityChangeBlock)block
+{
+    static dispatch_once_t once_t;
+    static dispatch_queue_t reachabilityQueue;
+    dispatch_once(&once_t, ^{
+        reachabilityQueue
+            = dispatch_queue_create("is.sentry.cocoa.connectivity", DISPATCH_QUEUE_SERIAL);
+    });
+
+    sentry_reachability_change_block = block;
+
+    const char *nodename = URL.host.UTF8String;
+    if (!nodename || ![self isValidHostname:@(nodename)]) {
+        return;
+    }
+
+    sentry_reachability_ref = SCNetworkReachabilityCreateWithName(NULL, nodename);
```

Comment:
Were you able to resolve the supposed memory leaks? I found this thread: https://github.com/tonymillion/Reachability/issues/115

Reply:
We don't have any memory leaks, nothing is triggered by the Analyzer. We're not using tonymillion/Reachability any more :)